docs/avro-data-source-guide.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: global
+title: Apache Avro Data Source Guide
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+Since Spark 2.4 release, [Spark SQL](https://spark.apache.org/docs/latest/sql-programming-guide.html) provides built-in support for reading and writing Apache Avro data.
+
```

Comment:
note I think we should add a compatibility section here, reference https://github.com/apache/spark/pull/22133

Reply:
@tgravescs I have add an independent section for it :)